### ISSUE TITLE
menuitemradio/menuitemcheckbox shouldn't support aria-expanded. Maybe radio should?

### ISSUE THREAD

**Issue by carmacleod** [w3c/aria#1277]
ARIA 1.2 issue https://github.com/w3c/aria/issues/681 and its PR https://github.com/w3c/aria/pull/972 revisited all roles that supported `aria-expanded`, and then:
- removed support for aria-expanded from grouping/container roles (i.e. the controlled element)
- added support for aria-expanded to checkbox (and switch, which inherits from checkbox)

However, a couple of things fell through the cracks.

#### Part 1. `menuitemcheckbox` and `menuitemradio` should not support `aria-expanded`
Currently menuitemcheckbox and menuitemradio inherit [aria-expanded](https://w3c.github.io/aria/#aria-expanded) from menuitem (and checkbox). 

We all agreed earlier that [menuitem needs to support aria-expanded](https://github.com/w3c/aria/issues/454) in order to allow for submenus, but menuitemcheckbox and menuitemradio are leaf nodes in a menu, so aria-expanded should be prohibited on them.

Looking at https://github.com/w3c/aria/issues/681#issuecomment-358731813 and https://github.com/w3c/aria/issues/681#issuecomment-488776046, it seems pretty clear that we intended to make this happen for ARIA 1.2.

#### Part 2. maybe `radio` should support `aria-expanded` now that `checkbox` does
Neither https://github.com/w3c/aria/issues/681 nor https://github.com/w3c/aria/pull/972 mention `radio`, and so even though it was decided that `checkbox` should support `aria-expanded`, it's not clear whether `radio` was considered as well.

People are using disclosures on radios (with or without aria-expanded).

The [gov.uk design system has a pattern with conditionally revealed content between radios](https://design-system.service.gov.uk/components/radios/#conditionally-revealing-content) ([example](https://design-system.service.gov.uk/components/radios/conditional-reveal/index.html)), and they give this advice:
> Using this component, you can add conditionally revealing content to stacked radios, so users only see content when it’s relevant to them.
> For example, you could reveal an email address input only when a user chooses to be contacted by email.

[PatternFly has this guideline](https://www.patternfly.org/v4/design-guidelines/usage-and-behavior/forms#use-progressive-disclosure):
> Use progressive disclosure to hide and show fields based on the users’ selection. This approach keeps the workflow short and allows the user to focus on only the fields relevant to them. It also avoids using color to indicate whether or not a field is available, making it suitable for visually impaired users.

In addition to forms, I've seen this used in print dialogs: (all pages, current page, custom + textbox) and in the controls for charts or transaction lists (last 5 days, last month, last 6 months, custom + datepicker).

The [gov.uk people did some testing of this with screen readers](https://github.com/alphagov/govuk-frontend/issues/979#issuecomment-528932826) (because it was flagged by a validator), and they discovered that there is enough AT support for aria-expanded on radios that they decided to keep using it in their forms, given that user testing did not find it problematic. In fact, they had some feedback that indicated that it is helpful to have expanded/collapsed announced.

Looking at the gov.uk test results, I see that NVDA/FF, VO/Safari and VO/iOS Safari all announce expanded/collapsed. I have just verified these, so it looks like we have our 2 implementations.

**Comment by 36degrees** [comment]
Hi from the Design System team at GOV.UK 👋🏻 

I was just wondering if Part 2 of this issue has been resolved? The linked PR #1278 which closed this out seems to be focused around part 1 (`menuitemcheckbox` and `menuitemradio`).

> Neither #681 nor #972 mention radio, and so even though it was decided that checkbox should support aria-expanded, it's not clear whether radio was considered as well.

For the reasons you outlined, we'd be interested to know if `radio` was previously considered, and what the outcome was. If radio was not considered, would it be possible to do so?

#1026 seems related – would that be a good place to continue this discussion?

**Comment by 36degrees** [comment]
@carmacleod it looks like originally #1278 [was going to extend support for `aria-expanded` to radios](https://github.com/w3c/aria/pull/1278#:~:text=Adds%20support%20for%20aria-expanded%20to%20radio), but after a call [the decision was made not to do this](https://github.com/w3c/aria/pull/1278#:~:text=don't%20support%20aria-expanded%20on%20radio). Is that correct?

Do you happen to remember any of the context for that decision? Was it just that the change was considered out of scope for the PR, or was there a decision made that radios should not support `aria-expanded`?

**Comment by jnurthen** [comment]
@36degrees The minutes are here - https://www.w3.org/2020/05/28-aria-minutes.html#item08
There was general disagreement about supporting expanded on radios. If you think it is important please log a new issue including specific use cases.
